### PR TITLE
Add aggregator_v2 related concurrent changes in token v2

### PR DESCRIPTION
**Pull request by vusirikala**
AIP 43 adds concurrency primitives in Token v2 standard as specified in https://github.com/aptos-foundation/AIPs/blob/main/aips/aip-43.md#impact. Adding these concurrency primitives in indexer as well. 

Changes addressed:
- Add ConcurrentSupply
- Add ConcurrentMintEvent
- Add ConcurrentBurnEvent
- Add ConcurrentTokenIdentifiers

# Testing
## ConcurrentSupply
Before
image
After
image

## ConcurrentIdentifier
Before
image

After
image

## ConcurrentMintEvent
Before
empty

After
image

### CODE REVIEW

File: rust/processor/src/models/token_v2_models/v2_token_utils.rs [hunk 1]
```diff
@@ -412,27 +495,71 @@ impl PropertyMapModel {
     }
 }
 
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct AggregatorSnapshot<T> {
+    // TODO: #[serde(deserialize_with = "deserialize_from_string")] is showing error
+    value: T,
+}
```
Comment (bowenyang007) [inline rust/processor/src/models/token_v2_models/v2_token_utils.rs:502]: wait so does this actually work?

Comment (vusirikala) [reply rust/processor/src/models/token_v2_models/v2_token_utils.rs:502]: Is the `serde` annotation required for the code to work?

File: rust/processor/src/models/token_v2_models/v2_token_activities.rs [hunk 2]
```diff
@@ -158,11 +158,19 @@ impl TokenActivityV2 {
...
+                V2TokenEvent::ConcurrentBurnEvent(inner) => inner.get_token_address(),
                 V2TokenEvent::TransferEvent(inner) => inner.get_object_address(),
                 _ => event_account_address.clone(),
             };
 
+            let event_type_override = match event_type.as_str() {
+                "0x4::collection::ConcurrentMintEvent" => "0x4::collection::MintEvent".to_string(),
+                "0x4::collection::ConcurrentBurnEvent" => "0x4::collection::BurnEvent".to_string(),
+                _ => event_type.clone(),
+            };
```
Comment (bowenyang007) [inline rust/processor/src/models/token_v2_models/v2_token_activities.rs:172]: yup this is what we want but may I suggest merging this with the match block below.

File: rust/processor/src/models/token_v2_models/v2_token_activities.rs [hunk 3]
```diff
@@ -173,6 +181,13 @@ impl TokenActivityV2 {
                         before_value: None,
                         after_value: None,
                     },
+                    V2TokenEvent::ConcurrentMintEvent(_) => TokenActivityHelperV2 {
+                        from_address: Some(object_core.get_owner_address()),
```
Comment (bowenyang007) [inline rust/processor/src/models/token_v2_models/v2_token_activities.rs:185]: We can do the event type override here.

File: rust/processor/src/models/token_v2_models/v2_token_activities.rs (hunk 2 above)
Comment (vusirikala) [reply rust/processor/src/models/token_v2_models/v2_token_activities.rs]: Done.

File: rust/processor/src/models/token_v2_models/v2_token_activities.rs (hunk 3 above)
Comment (vusirikala) [reply rust/processor/src/models/token_v2_models/v2_token_activities.rs:183]: Done.

**Review by bowenyang007 — APPROVED** [review]

File: rust/processor/src/models/token_v2_models/v2_collections.rs [hunk 4]
```diff
@@ -122,6 +122,16 @@ impl CollectionV2 {
                     );
                 }
 
+                // Aggregator V2 enables a separate struct for supply
+                let concurrent_supply = metadata.concurrent_supply.as_ref();
+                if let Some(supply) = concurrent_supply {
+                    (current_supply, max_supply, total_minted_v2) = (
+                        supply.current_supply.value.clone(),
+                        Some(supply.current_supply.max_value.clone()),
```
Comment (igor-aptos) [inline rust/processor/src/models/token_v2_models/v2_collections.rs:130]:
```
should we do here 

if supply.current_supply.max_value == u64::MAX { // or however BigDecimal and u64 can be compared
  None
} else {
  Some(supply.current_supply.max_value.clone())
}
```

**Review by igor-aptos — APPROVED** [review]

File: rust/processor/src/models/token_v2_models/v2_collections.rs (hunk 4 above)
Comment (vusirikala) [reply rust/processor/src/models/token_v2_models/v2_collections.rs]: Done.

**Pull request merged**